lib/net/pool.js

```diff
@@ -704,6 +729,60 @@ class Pool extends EventEmitter {
     this.compactBlocks.clear();
   }
 
+  /**
+   * Start the filters headers sync.
+   */
+
+  async startFilterHeadersSync() {
+    this.filterSyncing = true;
+    this.logger.info('Starting filter headers sync (%s).',
+      this.chain.options.network);
+    if (!this.opened || !this.connected)
+      return;
+
+    const cFHeaderHeight = this.cfHeaderChain.tail.height;
+    const startHeight = cFHeaderHeight
+    ? cFHeaderHeight + 1 : 1;
+    const chainHeight = await this.chain.height;
+    const stopHeight = chainHeight - startHeight + 1 > 2000
+     ? startHeight + 1999 : chainHeight;
+    const stopHash = await this.chain.getHash(stopHeight);
+    this.requestedFilterType = common.FILTERS.BASIC;
+    this.requestedStopHash = stopHash;
+    await this.peers.load.sendGetCFHeaders(
+      common.FILTERS.BASIC,
+      startHeight,
+      stopHash);
+  }
+
+  /**
+   * Start the filters sync.
+   */
+
+  async startFilterSync() {
+    this.logger.info('Starting filter sync (%s).',
+      this.chain.options.network);
+    if (!this.opened || !this.connected)
+      return;
+
+    const indexer = this.getFilterIndexer(filtersByVal[common.FILTERS.BASIC]);
+
+    const cFilterHeight = indexer.height;
+    const startHeight = cFilterHeight
+    ? cFilterHeight + 1 : 1;
+    const chainHeight = await this.chain.height;
+    const stopHeight = chainHeight - startHeight + 1 > 1000
+     ? startHeight + 999 : chainHeight;
+    const stopHash = await this.chain.getHash(stopHeight);
+    this.requestedFilterType = common.FILTERS.BASIC;
+    this.getcfiltersStartHeight = startHeight;
+    this.requestedStopHash = stopHash;
```

Comment:
I still think these should not be set at the pool class level. See `pool.resolveBlock()` and `pool.resolveTX()` for an idea of what I'm thinking: we might send different requests to different peers and that should be ok. I'm playing with the code now and getting a lot of "wrong stophash" warnings